include/triton/Analysis/Allocation.h

```diff
@@ -151,15 +166,26 @@ class Allocation {
     size_t size;
     size_t alignment;
     size_t offset;
+    SmallVector<BufferT *> aliases;
 
     bool operator==(const BufferT &other) const { return id == other.id; }
     bool operator<(const BufferT &other) const { return id < other.id; }
 
-    BufferT() : BufferT(BufferKind::Explicit, 0) {}
-    BufferT(BufferKind kind, size_t size, size_t alignment = 4,
+    BufferT() : BufferT(BufferKind::Explicit, 0, 0) {}
+    BufferT(BufferKind kind, BufferId id, size_t size, size_t alignment = 4,
             size_t offset = 0)
-        : kind(kind), id(nextId++), size(size), alignment(alignment),
-          offset(offset) {}
+        : kind(kind), id(id), size(size), alignment(alignment), offset(offset) {
+    }
+
+    void apply(const std::function<void(BufferT *)> &f) {
```

Comment:
`apply`->`applyToBufferAndAliases`?

Reply:
changing to `applyToActuals`? and making it recursive since Alias may refer to Alias (and handle cycles).